# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/WebIdentityHandler.h
# Commit: 459739221a88
# Full Hash: 459739221a884b2b1c2adab42fd19fafe279bd20
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing build bustages
#   
#   This reverts commit ab681e36aeafc21ba62bb4680bdaa6206f02546d.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r fc9986230203 -r 459739221a88 dom/credentialmanagement/identity/WebIdentityHandler.h
--- a/dom/credentialmanagement/identity/WebIdentityHandler.h	Thu Jun 05 00:15:19 2025 +0300
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,55 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef mozilla_dom_WebIdentityHandler_h
-#define mozilla_dom_WebIdentityHandler_h
-
-#include "mozilla/dom/AbortFollower.h"
-#include "mozilla/dom/CredentialManagementBinding.h"
-#include "mozilla/dom/WebIdentityChild.h"
-#include "nsCycleCollectionParticipant.h"
-#include "nsPIDOMWindow.h"
-
-namespace mozilla::dom {
-
-class WebIdentityHandler final : public AbortFollower {
- public:
-  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
-  NS_DECL_CYCLE_COLLECTION_CLASS(WebIdentityHandler)
-
-  explicit WebIdentityHandler(nsPIDOMWindowInner* aWindow) : mWindow(aWindow) {
-    MOZ_ASSERT(NS_IsMainThread());
-    MOZ_ASSERT(aWindow);
-  }
-
-  bool MaybeCreateActor();
-  void ActorDestroyed();
-
-  // AbortFollower
-  void RunAbortAlgorithm() override;
-
-  void GetCredential(const CredentialRequestOptions& aOptions,
-                     bool aSameOriginWithAncestors,
-                     const RefPtr<Promise>& aPromise);
-
-  void PreventSilentAccess(const RefPtr<Promise>& aPromise);
-
-  void Disconnect(const IdentityCredentialDisconnectOptions& aOptions,
-                  const RefPtr<Promise>& aPromise);
-
-  void SetLoginStatus(const LoginStatus& aStatus,
-                      const RefPtr<Promise>& aPromise);
-
- private:
-  ~WebIdentityHandler();
-  nsCOMPtr<nsPIDOMWindowInner> mWindow;
-  RefPtr<WebIdentityChild> mActor;
-  RefPtr<Promise> mGetPromise;
-};
-
-}  // namespace mozilla::dom
-
-#endif  // mozilla_dom_WebIdentityHandler_h